I3uckwheat/cards-of-carousal
# Feature/222 respect player count setting

#### 0. Make sure your issue is linked:

closes #222 


#### 1. Purpose:

This addition is to respect the maxPlayer limit set by the host. If a player tries to join after the limit has been reached, they will receive a message. 


#### 2. Implementation:

I handled this mostly in `HostReducer` by checking the amount of current players before adding a new player to the host's state - from there I pulled in `hostReducerMiddleware` and can send the "player limit reached" message if needed.

#### 3. Possible Issues:

- We may want to refactor how the player connects by removing the `PLAYER_CONNECTED` case from `hostReducerMiddleware` and instead, pulling the middleware into the reducer and sending the message from there. With the way I'm currently kicking players that have exceeded the max player limit, the player is still being sent the "player connected" message even when they are too many players. We're just sending another message after to kick them. It's unnoticeable to the user but may be an anti-pattern.
- We might want to rename `sendTooManyPlayersMessage()`?
- Potential issue with mocking `hostReducerMiddleware` in `HostReducer.spec.js`
- More test cases might be needed


#### 4. How to Test:


Test that players cannot join after limit has been reached
- run frontend and backend
- open 4 tabs
- in the first tab, host a lobby and set the max player limit to 2
- in the second and third tabs, join the lobby
- in the fourth tab, try to join the lobby and you should see the "player limit reached" screen
- change the player limit to 3 and join the lobby in the 4th tab
- in the host, while 3 players are joined, try to lower the player limit to 2

#### 5. Cleanup
- [x ] I've added all TODOs needed
- [ x] I've removed all unneeded comments
- [ x] I've updated the snapshot tests and checked to make sure they're accurate if they changed
- [ x] I've removed all unnecessary `console.log`s
